src/lib/components/DeleteButton.js

```diff
@@ -50,10 +51,11 @@ export class DeleteButtonComponent extends Component {
       isVersion,
       ...uiProps // only has ActionButton props
     } = this.props;
-
+    const { isDeleting } = this.state;
     const handleDelete = (event, formik) => {
       deleteClick(event, formik);
       this.handleClose();
+      this.setState({ isDeleting: true });
     };
```

Comment:
:detective: : This actually reveals a pre-exisitng problem: we are not dealing with errors in deletion correctly. Clearly here `isDeleting` is set to true forever if the delete didn't work, but the problem is larger: `formik` is not a `formik` object at all! It's a `data: object` containing all the props of `Button` ([reference](https://react.semantic-ui.com/elements/button/#types-button)). `Button` [here](https://github.com/inveniosoftware/react-invenio-deposit/pull/258/files#diff-e76813af199e88111670c54019b3d3c5f7f732f8e8f7329d6ad0d069c133d7c3R107) should be an `ActionButton` to pass a formikBag. It needs more investigation to even see if we need formikBag in this case at all. 

I made a separate issue about all this: https://github.com/inveniosoftware/react-invenio-deposit/issues/259 .

Thanks for triggering this investigation.

Reply:
I see, I added the `isDeleting` and set it true forever because I saw this comment [here](https://github.com/inveniosoftware/react-invenio-deposit/blob/master/src/lib/DepositController.js#L180-L182) and assumed all good for the moment.